### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by bdice**
## Description
Makes the package more easily maintainable by any signac committer / maintainer.

## Checklist:
- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac-dashboard/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac-dashboard/blob/master/ContributorAgreement.md).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac-dashboard/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-dashboard/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

**Comment by tommy-waltmann** [comment]
Will the changes in who is requested for review show up on this PR, or do we need to wait until this is merged before really knowing if the changes are working right?

**Review by csadorf — APPROVED** [review]

**Comment by bdice** [comment]
@tommy-waltmann CODEOWNERS is pulled from the default branch. The changes take effect when this is merged.

**Pull request merged**
